### ISSUE TITLE
PrefixExpression incorrect range/location

### ISSUE THREAD

**Issue by trixnz** [oxyc/luaparse#47]
`parsePrefixExpression` seems to be incorrectly handling the propagation of child ranges and locations, as demonstrated with the following Lua snippet:
```lua
base['outer']['inner']()
```

### Expected behaviour
The outer `IndexExpression`'s range should contain the inner expression without the `CallExpression`'s parenthesis and the inner expression should only contain its own range.

#### Output
```json
{
  "type": "Chunk",
  "body": [
    {
      "type": "CallStatement",
      "expression": {
        "type": "CallExpression",
        "base": {
          "type": "IndexExpression",
          "base": {
            "type": "IndexExpression",
            "base": {
              "type": "Identifier",
              "name": "base",
              "range": [
                0,
                4
              ]
            },
            "index": {
              "type": "StringLiteral",
              "value": "outer",
              "raw": "'outer'",
              "range": [
                5,
                12
              ]
            },
            "range": [
              13,
              22
            ]
          },
          "index": {
            "type": "StringLiteral",
            "value": "inner",
            "raw": "'inner'",
            "range": [
              14,
              21
            ]
          },
          "range": [
            0,
            22
          ]
        },
        "arguments": [],
        "range": [
          0,
          24
        ]
      },
      "range": [
        0,
        24
      ]
    }
  ],
  "range": [
    0,
    24
  ],
  "comments": []
}
```

### Actual behaviour
Both `IndexExpression`s refer to the same range as the entire `CallExpression` (including its `()`, which is incorrect)

#### Output
```json
{
  "type": "Chunk",
  "body": [
    {
      "type": "CallStatement",
      "expression": {
        "type": "CallExpression",
        "base": {
          "type": "IndexExpression",
          "base": {
            "type": "IndexExpression",
            "base": {
              "type": "Identifier",
              "name": "base",
              "range": [
                0,
                4
              ]
            },
            "index": {
              "type": "StringLiteral",
              "value": "outer",
              "raw": "'outer'",
              "range": [
                5,
                12
              ]
            },
            "range": [
              0,
              24
            ]
          },
          "index": {
            "type": "StringLiteral",
            "value": "inner",
            "raw": "'inner'",
            "range": [
              14,
              21
            ]
          },
          "range": [
            0,
            24
          ]
        },
        "arguments": [],
        "range": [
          0,
          24
        ]
      },
      "range": [
        0,
        24
      ]
    }
  ],
  "range": [
    0,
    24
  ],
  "comments": []
}
```

**Comment by fstirlitz** [comment]
Fixed by fc6fdf4025a96b423655e4496f236689938a26be, thank you.

**Comment by trixnz** [comment]
Thanks for the quick turn around! This cascaded down and fixed a whole slew of hacks I had to implement in my formatter. Appreciate it! 😄
